packages-exp/analytics-exp/src/constants.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Type constant for Firebase Analytics.
+ */
+export const ANALYTICS_TYPE = 'analytics-exp';
+
+// Key to attach FID to in gtag params.
+export const GA_FID_KEY = 'firebase_id';
+export const ORIGIN_KEY = 'origin';
+
+export const FETCH_TIMEOUT_MILLIS = 60 * 1000;
+
+export const DYNAMIC_CONFIG_URL =
+  'https://firebase.googleapis.com/v1alpha/projects/-/apps/{app-id}/webConfig';
+
+export const GTAG_URL = 'https://www.googletagmanager.com/gtag/js';
+
+export enum GtagCommand {
+  EVENT = 'event',
+  SET = 'set',
+  CONFIG = 'config'
+}
+
+/*
+ * Officially recommended event names for gtag.js
+ * Any other string is also allowed.
+ */
+export const enum EventName {
```

Comment:
I think we wanted to make the enum available to users so they could easily get a list of official EventNames that they can then put into `logEvent`. The string literals are already available as autocompletes because of the overloads but maybe they want to save it to a variable? Feel safer from typos?

It wasn't exported so I added it to exports from `api.ts`.

Basically, if you import the `EventName` enum you can do:

`logEvent(analytics, EventName.LOGIN, { method: 'email' });`

or

`logEvent(analytics, 'login', { method: 'email' });`

Reply:
If we want to export it, it has to be a normal enum instead of const enum because const enum only exists in compile time. Is it part of the API proposal? I'm okay exporting it since it's useful, but I don't think enums are tree-shakable, so it will be included in bundle even if not used.

Reply:
I think we should keep it especially as it's available in the current API. I'll add it to the API proposal and send an email.